File: Python/CS001-week06/hanoi.py
def hanoi(n,x,y,z):
    if n==1:
        print("Move disk",1,"from",x,"to",z)
    else:
        hanoi(n-1,x,z,y)
        print("Move disk",n,"from",x,"to",z)
        hanoi(n-1,y,x,z)

hanoi(3,"A","B","C")

File: Python/CS001-week08/Dev-C++ 6.0 更名为SmartC++/Examples/OI/Low/Hanoi.cpp
#include<stdio.h>
void hanoi(int n, char A, char B, char C) {
    if(n == 1) {
    printf("Move disk %d from %c to %c\n", n, A, C);
    } else {
        hanoi(n - 1, A, C, B);
        printf("Move disk %d from %c to %c\n", n, A, C);
        hanoi(n - 1, B, A, C);
    }
}
int main(void) {
    int n;
    printf("请输入数字n以解决n阶汉诺塔问题：\n");
    scanf("%d", &n);
    hanoi(n, 'A', 'B', 'C');
}
